src/mlpack/methods/ann/rbm/rbm.hpp

```diff
@@ -118,7 +120,7 @@ class RBM
    */
   template<typename Policy = PolicyType>
   typename std::enable_if<std::is_same<Policy, BinaryRBM>::value, double>::type
-  FreeEnergy(arma::Mat<ElemType>&& input);
+  FreeEnergy(const arma::Mat<ElemType>& input);
```

Comment:
On a first look I think we could also add `InputType` as a function template instead of a class template, that would also mean we don't necessary break backward compatibilty. So that would mean `template<typename Policy = PolicyType, typename InputType = DataType>`, let me know what you think.

Reply:
Thanks @zoq I changed as suggested.

Reply:
Hey @zoq I don't know but the windows build was failing in previous builds may be windows compiler is treating `InputType` and `DataType` as same. As we have already discussed and it is happening after changing as you suggested.

Reply:
Hm, does it work if you put `InputType` first and without the default type?

Reply:
It worked I introduced New template parameter `InputType` for RBM class.

Reply:
Hmm, strange Windows build pass now I don't why these mysterious error's always occur on my prs :grin:

Reply:
Great that you solved it.